src/testing/jest/README.md

```diff
@@ -0,0 +1,210 @@
+    +++ b/src/testing/jest/jest-28/package-lock.json
+    @@ -1,11 +1,11 @@
+    {
+    -  "name": "@stencil/jest-27-and-under",
+    +  "name": "@stencil/jest-28",
+       "version": "1.0.0",
+       "lockfileVersion": 3,
+       "requires": true,
+       "packages": {
+       "": {
+    -      "name": "@stencil/jest-27-and-under",
+    +      "name": "@stencil/jest-28",
+           "version": "1.0.0",
+           "license": "MIT",
+           "devDependencies": {
+    diff --git a/src/testing/jest/jest-28/package.json b/src/testing/jest/jest-28/package.json
+    index 92b5ee419..bd1614fe0 100644
+    --- a/src/testing/jest/jest-28/package.json
+    +++ b/src/testing/jest/jest-28/package.json
+    @@ -1,5 +1,5 @@
+    {
+    -  "name": "@stencil/jest-27-and-under",
+    +  "name": "@stencil/jest-28",
+       "version": "1.0.0",
+       "description": "Internal package for supporting Jest 27 and under with Stencil",
+       "license": "MIT",
+    
+    ```
+1. Update the description of the package in `package.json` to match the supported version of Jest:
+    ```diff
+    --- a/src/testing/jest/jest-28/package.json
+    +++ b/src/testing/jest/jest-28/package.json
+    @@ -1,7 +1,7 @@
+     {
+       "name": "@stencil/jest-28",
+       "version": "1.0.0",
+    -  "description": "Internal package for supporting Jest 27 and under with Stencil",
+    +  "description": "Internal package for supporting Jest 28 with Stencil",
+       "license": "MIT",
+       "author": "Ionic Team",
+       "homepage": "https://stenciljs.com/",
+    
+    ```
+1. Update `renovate.json` to not bump Jest beyond the version that this directory is responsible for:
+    ```diff
+    index 4861b2477..7cf75e16d 100644
+    --- a/renovate.json5
+    +++ b/renovate.json5
+    @@ -98,6 +98,11 @@
+           matchPackageNames: ['@types/jest', 'jest'],
+           allowedVersions: '<=27'
+         },
+    +    {
+    +      matchFileNames: ["src/testing/jest/jest-28/package.json"],
+    +      matchPackageNames: ['@types/jest', 'jest'],
+    +      allowedVersions: '<=28'
+    +    },
+         {
+    
+    ```
+   Note how:
+   - `matchFileNames` maps to the new Jest directory's `package.json`
+   - `matchPackageNames` lists the packages that were upgraded earlier
+   - `allowedVersions` allows versions less than or equal to the version we're adding support for
+1. Add the new version of Jest to the Component Starter smoke tests:
+    ```diff
+   index 16560a659..a4bee5308 100644
+    --- a/.github/workflows/test-component-starter.yml
+    +++ b/.github/workflows/test-component-starter.yml
+    @@ -16,7 +16,7 @@ jobs:
+    strategy:
+    fail-fast: false
+    matrix:
+    -        jest: ['24', '25', '26', '27']
+    +        jest: ['24', '25', '26', '27', '28']
+             node: ['16', '18', '20']
+             os: ['ubuntu-latest', 'windows-latest']
+      runs-on: ${{ matrix.os }}
+   ```
+1. Rename the `JestFacade` implementation in the new directory's `jest-facade.ts` file.
```

Comment:
This step could be a little clearer that you need to rename the class in `src/testing/jest/jest-<version>/jest-facade.ts` and also its references in `src/testing/jest/jest-<version>/jest-config.ts`

Reply:
I'm not sure I follow this - wouldn't [this line](https://github.com/ionic-team/stencil/pull/4952/files#diff-2c531f2d33dab017a4385abe8baffc2b3f78067802baf95f257f4f4b5deee42eR125) cover it? 

> It is recommended that you use your editor's renaming/refactoring utility to do this step:

Maybe it's different for each editor, but I think IntelliJ (what I use/my basis for this statement) does that. I don't think we have any usages of any Facade implementation in our `tests/` either, so I'm not sure how to best act on this one 🤔

Reply:
Oh yeah, I guess I didn't realize that the editors would rename the symbol in other files that referenced it with an import

Reply:
fwiw any editor that uses the typescript language server should be able to do this (I can do that in neovim with it)